xarray/core/duck_array_ops.py

```diff
@@ -171,6 +171,87 @@ def _ignore_warnings_if(condition):
         yield
 
 
+def _nansum(value, axis=None, **kwargs):
+    """ In house nansum. This is used for object array """
+    value = fillna(value, 0.0)
+    return _dask_or_eager_func('sum')(value, axis=axis, **kwargs)
+
+
+def _nan_minmax(func, fill_value, value, axis=None, **kwargs):
+    """ In house nan-reduce used for nanmin, nanmax. This is used for object
+    array """
+    valid_count = count(value, axis=axis)
+    filled_value = fillna(value, fill_value)
+    data = _dask_or_eager_func(func)(filled_value, axis=axis, **kwargs)
+    if not hasattr(data, 'dtype'):  # scalar case
+        data = np.nan if valid_count == 0 else data
+        return np.array(data, dtype=value.dtype)  # return 0d-array
+    # convert all nan part axis to nan
+    return where_method(data, valid_count != 0)
+
+
+def _nan_argminmax(func, fill_value, value, axis=None, **kwargs):
+    """ In house nan-reduce used for nanargmin, nanargmax. This is used for
+    object array. This always return integer type """
+    valid_count = count(value, axis=axis)
+    value = fillna(value, fill_value)
+    data = _dask_or_eager_func(func)(value, axis=axis, **kwargs)
+    # dask seems return non-integer
+    if isinstance(value, dask_array_type):
+        data = data.astype(int)
+    if not hasattr(data, 'dtype'):  # scalar case
+        # TODO should we raise ValueError if all-nan slice encountered?
+        data = -1 if valid_count == 0 else int(data)
+        return np.array(data)  # return 0d-array
+    # convert all nan part axis to nan
+    return where_method(data, valid_count != 0, -1)
```

Comment:
In numpy, `nanargmin` raise ValueError if there are all-NaN slice/axis.
Do we follow this?
Considering our `getitem_with_mask` method, I think it is consistent to return -1 for such a case, 
but it could be confusing sometimes.